server/src/main/java/org/elasticsearch/index/mapper/MappingParser.java

```diff
@@ -118,10 +120,14 @@ private Mapping parse(String type, Map<String, Object> mapping) throws MapperPar
                 @SuppressWarnings("unchecked")
                 Map<String, Object> fieldNodeMap = (Map<String, Object>) fieldNode;
                 MetadataFieldMapper metadataFieldMapper = typeParser.parse(fieldName, fieldNodeMap, parserContext)
-                    .build(MapperBuilderContext.ROOT);
+                    .build(MapperBuilderContext.root(false));
                 metadataMappers.put(metadataFieldMapper.getClass(), metadataFieldMapper);
                 fieldNodeMap.remove("type");
                 checkNoRemainingFields(fieldName, fieldNodeMap);
+
+                if (metadataFieldMapper instanceof SourceFieldMapper sfm) {
+                    isSourceSynthetic = sfm.isSynthetic();
+                }
             }
```

Comment:
Yeah. I was thinking maybe metadata fields need to get a version that throws an exception if you check if synthetic source is enabled. Or maybe get an object that doesn't have that method. Not sure which is more sane.

Reply:
@romseygeek and I talked in another channel and figured out that the metadata field `_doc_count` needs to know if synthetic source is enabled. Well, it will when I add support for synthetic _source to it. Which I should do! So I think we should merge this as is and I'll add support for doc count on top of it, messing with this code right here to make the synthetic _source-ness available to metadata mappers.